shared/src/utils/propositionalLogicMinimize.ts

```diff
@@ -0,0 +1,324 @@
+import {
+  BinaryOperatorType,
+  Literal,
+  NormalForm,
+  Operator,
+  SyntaxTreeNodeType,
+  TruthTable,
+} from "@shared/utils/propositionalLogic.ts"
+
+/**
+ * During the Quine–McCluskey algorithm, the datatype for a truthTable can be a BinaryMix
+ * 0 --> false; 1 --> true; x --> true/false
+ */
+type BinaryMix = 0 | 1 | "x"
+
+/**
+ * This class gets the minimal normal form of a given expression.
+ * - DNF (Disjunctive Normal Form)
+ * - CNF (Conjunctive Normal
+ * - The minimal normal form is the minimal number of literals needed
+ */
+export class GetMinimalNormalForm {
+  private readonly expr: SyntaxTreeNodeType
+  private readonly cnfExpr: SyntaxTreeNodeType
+  private readonly dnfExpr: SyntaxTreeNodeType
+
+  constructor(expr: SyntaxTreeNodeType) {
+    this.expr = expr
+    this.dnfExpr = this.getMinimalForm("DNF")
+    this.cnfExpr = this.getMinimalForm("CNF")
```

Comment:
If I see it correctly, only one of these two are every used in the end.
So maybe remove the `readonly` tag and compute the result in the `get` function. You can store the result in the respective class property to avoid recomputing it.
Or you ass the form type to the constructor. This would also remove the need to resupply the form type and you only loose the ability to use the object for both types of normal forms, which is not used anyway. It would also combine the two properties (`dnfExpr` and `cnfExpr`) to a generic one.

Reply:
Okay, I added the form to the constructor and only `this.minExpr` exists now. But this will still be computed in the constructor

Reply:
yes, that is fine.